src/DeviceManager.Api/Controllers/DevicesController.cs

```diff
@@ -8,8 +8,11 @@
 
 namespace DeviceManager.Api.Controllers
 {
+    /// <summary>
+    /// CRUD operations of the Device 
+    /// </summary>
     [Route("api/v{version:apiVersion}/devices")]
-    public class DevicesController : Controller
+    public class DevicesController : BaseController<DeviceViewModel>
```

Comment:
Nice, Just wondering, why base controller uses Device View Model ?

Reply:
I thought we can push common crud operations to BaseController (Should be an abstract class), so that other controller no need to write same code once again.

Reply:
Having base controller this is, definitely, cool feature. Thank you. However I think it doesn't intended for it because:  
- All methods in controller are unique (or should be) and all methods/api calls have a version
- And API controller this is just a web interface to your Business/Domain logic 
- For CRUD operation we have Data Access layer and generic Repository already have common logic for Create, Update, Delete. 

According to my own experience. ;)  

But anyway thank you a lot, I will take this changes and merge this PR.

Reply:
Yeah, you are right :+1: